qiskit/primitives/base_estimator.py

```diff
@@ -74,53 +74,50 @@
     H2 = SparsePauliOp.from_list([("IZ", 1)])
     H3 = SparsePauliOp.from_list([("ZI", 1), ("ZZ", 1)])
 
-    with Estimator([psi1, psi2], [H1, H2, H3], [params1, params2]) as e:
-        theta1 = [0, 1, 1, 2, 3, 5]
-        theta2 = [0, 1, 1, 2, 3, 5, 8, 13]
-        theta3 = [1, 2, 3, 4, 5, 6]
-
-        # calculate [ <psi1(theta1)|H1|psi1(theta1)> ]
-        result = e([0], [0], [theta1])
-        print(result)
-
-        # calculate [ <psi1(theta1)|H2|psi1(theta1)>, <psi1(theta1)|H3|psi1(theta1)> ]
-        result2 = e([0, 0], [1, 2], [theta1]*2)
-        print(result2)
-
-        # calculate [ <psi2(theta2)|H2|psi2(theta2)> ]
-        result3 = e([1], [1], [theta2])
-        print(result3)
-
-        # calculate [ <psi1(theta1)|H1|psi1(theta1)>, <psi1(theta3)|H1|psi1(theta3)> ]
-        result4 = e([0, 0], [0, 0], [theta1, theta3])
-        print(result4)
-
-        # calculate [ <psi1(theta1)|H1|psi1(theta1)>,
-        #             <psi2(theta2)|H2|psi2(theta2)>,
-        #             <psi1(theta3)|H3|psi1(theta3)> ]
-        result5 = e([0, 1, 0], [0, 1, 2], [theta1, theta2, theta3])
-        print(result5)
-
-        # Objects can be passed instead of indices.
-        # calculate [ <psi2(theta2)|H2|psi2(theta2)> ]
-        # Note that passing objects has an overhead
-        # since the corresponding indices need to be searched.
-        result6 = e([psi2], [H2], [theta2])
-        print(result6)
+    theta1 = [0, 1, 1, 2, 3, 5]
+    theta2 = [0, 1, 1, 2, 3, 5, 8, 13]
+    theta3 = [1, 2, 3, 4, 5, 6]
+
+    estimator = Estimator([psi1], [H1], [params1])
+
+    # Specify the circuit and observable by indices.
+    # calculate [ <psi1(theta1)|H1|psi1(theta1)> ]
+    result = estimator.run([0], [0], [theta1]).result()
+    print(result)
+
+    # Objects can be passed instead of indices.
+    # Note that passing objects has an overhead
+    # since the corresponding indices need to be searched.
```

Comment:
Can't we use a dictionary keyed by the object id, then it'd be much faster than looping through the list (not that people are expected to have that many objects).

Reply:
Right. O(1)

Reply:
Supported in c0fcc57